data/simpy.gram

```diff
@@ -27,7 +27,7 @@ compound_stmt[stmt_ty]: function_def | if_stmt | class_def | with_stmt | for_stm
 
 # NOTE: yield_expression may start with 'yield'; yield_expr must start with 'yield'
 assignment:
-    | '(' a=ann_assign_target ')' ':' b=expression c=['=' d=yield_expression { d }] {
+    | '(' a=paren_ann_assign_target ')' ':' b=expression c=['=' d=yield_expression { d }] {
         _Py_AnnAssign(a, b, c, 0, EXTRA)}
     | a=ann_assign_target ':' b=expression c=['=' d=yield_expression { d }] {
         _Py_AnnAssign(a, b, c, 1, EXTRA) }
```

Comment:
These seem to be redundant except for the `0` or `1` argument to `_Py_AnnAssign()` -- and why do we need that? (I can't even figure out when simple=1 is ever set in an AST made by `ast.parse()`?)

Reply:
`simple=1` when there aren't any parentheses, `simple=0` otherwise.

In Python/ast.c there is the following snippet:

```
int simple = 1;
...
if (NCH(deep) > 0 && TYPE(CHILD(deep, 0)) == LPAR) {
    simple = 0;
}
```

Reply:
Hm, I tried this:
```
>>> ast.dump(ast.parse('(a):int'))
"Module(body=[AnnAssign(target=Name(id='a', ctx=Store()), annotation=Name(id='int', ctx=Load()), value=None, simple=0)], type_ignores=[])"
>>>
```
And this still comes up with `simple=0` (no matter how many parens I put around it).

Reply:
Right. And that's because there are parentheses around the NAME, hence it's not simple (`simple=0`). But if you do:

```
>>> t = 'a: int'
>>> ast.dump(ast.parse(t))
"Module(body=[AnnAssign(target=Name(id='a', ctx=Store()), annotation=Name(id='int', ctx=Load()), value=None, simple=1)], type_ignores=[])"
```

you get `simple=0`.

Reply:
D'oh. I had the sense of the flag backwards. :-(